### PR TITLE
refactor: create base classes

### LINKED ISSUE #78: Refactor to base classes
Towards #77. Create base classes for shared functionality between async and sync clients. Contains non-networked functions.

### PR DESCRIPTION
**Pull request by rafilong**
Resolves #78 

Commit 496f501 onwards describes refactor changes
Commit f7fa191 sets up refactor

### CODE REVIEW

**Comment by crwilcox** [comment]
https://github.com/googleapis/python-bigtable/pull/79/files/f7fa191da03d125254b2a7e9be8e5956d7124d67..3b9c72665ce6161c4ac5ce4922c5b764a0497a67 will get the range you likely want to review ;)

File: noxfile.py [hunk 1]
```diff
@@ -85,13 +85,13 @@ def default(session):
     )
 
 
-@nox.session(python=["2.7", "3.5", "3.6", "3.7"])
```
Comment (crwilcox) [inline noxfile.py:88]: can you also
- add 3.8, drop 3.5
- make the corresponding changes to setup.py

**Comment by rafilong** [comment]
As @crwilcox pointed out, this PR involves a lot of moving code. It most likely makes sense to review the existing file changes from the entire PR changes, and review the changes to new files from [here](https://github.com/googleapis/python-bigtable/pull/79/files/f7fa191da03d125254b2a7e9be8e5956d7124d67..3b9c72665ce6161c4ac5ce4922c5b764a0497a67).

My approach to authoring this PR was to move the original classes to the base class version and copy them back in a separate commit, so the commits from that point onwards mostly see code deletion.

File: google/cloud/bigtable/app_profile.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
+# Copyright 2018 Google LLC.
```
[the remark this answers is not in the document]
Comment (rafilong) [reply google/cloud/bigtable/app_profile.py]: Resolved in chat.

**Pull request merged**